synedrion/src/paillier/encryption.rs

```diff
@@ -113,7 +126,7 @@ impl<P: PaillierParams> CiphertextMod<P> {
     fn new_with_randomizer_inner(
         pk: &PublicKeyPaillierPrecomputed<P>,
         abs_plaintext: &P::Uint,
-        randomizer: &Randomizer<P>,
+        randomizer: SecretBox<Randomizer<P>>,
```

Comment:
It felt a bit silly to take a reference to a `Box`, but perhaps it was fine as it was before (the compiler will likely optimize away the double pointer?). Thought?

Reply:
Wouldn't it be better to have `Randomizer` impl `ZeroizeOnDrop`? I think it is always secret.

Reply:
This way you woudn't have to clone everywhere.

Reply:
It does! See [here](https://github.com/dvdplm/synedrion/blob/28b36e926b5998a20d36fa5f8f1fd937cfa8dab1/synedrion/src/paillier/encryption.rs#L18). And [`RandomizerMod`](https://github.com/dvdplm/synedrion/blob/28b36e926b5998a20d36fa5f8f1fd937cfa8dab1/synedrion/src/paillier/encryption.rs#L38) does as well.

(This was the case before this PR)

In my head I thought `SecretBox`ing values was orthogonal to zeroizing them; the former to try to ensure nothing is left on the stack and the latter to ensure data is zeroed out after use (regardless of stack/heap). But I guess I was wrong about that then. Maybe `ZeroizeOnDrop` is all we need, and then add `Debug` impls to redact the secrets?

Reply:
The way I see it, `SecretBox` is a wrapper for some types that may or may not be secret (e.g. a `Scalar`, or some byte array), so you add the zeroization to them externally if it's necessary. But if the type is already zeroized at all times, no `SecretBox` is needed.

Reply:
Ok, I reverted the changes that wrapped `Randomizer` and `RandomizerMod` in a `SecretBox` (which coincidentally made the `clone`s go away too).